lib/puppet/pops/types/type_calculator.rb

```diff
@@ -1192,22 +1201,34 @@ def tuple_entry_at(tuple_t, from, to, index)
   # @api private
   #
   def assignable_PStructType(t, t2)
-    return true if t == t2 || t.elements.empty? && (t2.is_a?(Types::PHashType))
-    h = t.hashed_elements
+    els = t.elements
+    return true if t == t2 || els.empty? && (t2.is_a?(Types::PHashType))
```

Comment:
I removed the special test since even if t2 is a Hash, it's key_type needs to be assignable to the Struct key_type (e.g. String). The normal test is quick enough anyway.